### DISCUSSION TITLE
How to query for the installed version?

### DISCUSSION THREAD

**Discussion by phampyk** [FreshRSS/FreshRSS#6731] [Q&A]
Hello, I've got a bit of a silly question... but I'm setting up a monitoring app for my self hosted services, and it can check .json for the installed version and compares it with the latest github releases. I was checking the FreshRSS API but it only has greader API and there you can't query for the installed version. 
I would appreciate any help if this is possible. Thank you!

**Comment by Alwaysin** [comment]
You can read the value of "const FRESHRSS_VERSION =" in constants.php file
Or you can use git if it's how you installed FreshRSS.

I guess the [CLI ](https://github.com/FreshRSS/FreshRSS/tree/edge/cli) could also be extended to have a command to return the version installed.

**Reply by phampyk** [reply]
I'm not sure I can use any of those. For context this is the monitoring app https://github.com/release-argus/Argus/ and I'm still learning about the ins and outs of using apis and stuff.

Is there a way to get to the constants.php with a GET Method. 

I installed freshrss with a Softaculous installation so I'm not sure what the installation method is, but I think is just the zip. 

I'm sorry for the trouble and the lack of knowledge, and thank you!

**Reply by Alwaysin** [reply]
I don't know about the specifics of this software, but if it can only do web requests, maybe you could simply query ```https://<your-freshrss-domain>/i/?c=update``` and parse it. But indeed if you do not have access to the underlying server, I guess you wont be able to use any of the three methods I have suggested above anyway.

**Reply by phampyk** [reply]
As far as I'm aware (as I said, I'm still learning about it) it can only do web requests and parse json. I'm not sure if `https://<your-freshrss-domain>/i/?c=update` can be parsed in json... I think not, right?

**Reply by Alwaysin** [reply]
In their demo config here: https://release-argus.io/demo/config you can see it can parse any webpage with a regex, see for example the config for ```golang/go```
And I've just noticed you can see the version of FreshRSS on any webpage, even not authenticated, there is a json included exposing the version:
`    "version": "1.24.2-dev"`. I don't know if this is desirable, @Alkarex ?

**Reply by phampyk** [reply]
The parsing any webpage is for checking when there's a new version, not the deployed version sadly. If not it would have been a lot easier for me.

Where would it be the exposed json file? What would the URL be?

**Reply by Alkarex** [reply]
@Alwaysin The reason for having the version number client side is for extensions https://github.com/FreshRSS/FreshRSS/issues/5542
We could discuss whether this is needed also when not logged in, though.

As for the parent question, we should try to find a better approach, and it will be easier when we move to an automated process also for ZIP-based installs (https://github.com/FreshRSS/FreshRSS/pull/1760 ; https://github.com/FreshRSS/update.freshrss.org/pull/13#issuecomment-2156059737)

**Reply by phampyk** [reply]
So for what I can gather it's not possible to do what I was looking for.

That's a bummer, but thank you for your answers anyway :)

**Reply by Alwaysin** [reply]
> So for what I can gather it's not possible to do what I was looking for.
> 
> That's a bummer, but thank you for your answers anyway :)

Have you tried it? As I said, there is a json provided on any freshrss webpage, you just have to query the login page. In their examples they do it for Grafana, they query the login page which contains a json containing the version, so I do not see why that wouldn't work here.

**Reply by phampyk** [reply]
The json you mention is inside a <script> tag, and as far as I can see argus cannot parse it. The only .json available that argus can parse is manifest.json, and that one doesn't have the version.

I'm seeing that argus is pretty limited when it's about parsing the deployed version, at least from my perspective as someone just learning.

**Reply by Alwaysin** [reply]
Yes and so is the json on a grafana webpage. Why would they have made it in their example if it would not work?
But, if you're not willing to try, we will indeed never know.

**Reply by phampyk** [reply]
OH MY!! You are right!!!! I managed using the example from grafana!! So simple and here I was making it so complicated. THANK YOU so much :D
